### PR TITLE
 pre_hide_files: iterate over stages, not parent images

### PR DESCRIPTION
**Pull request by twaugh**

### CODE REVIEW

File: atomic_reactor/plugins/pre_hide_files.py [hunk 1]
```diff
@@ -43,8 +43,28 @@ def run(self):
         self._populate_start_file_lines()
         self._populate_end_file_lines()
 
-        for parent_image_id in self.workflow.builder.parent_images.values():
-            self._update_dockerfile(parent_image_id)
+        # Find limits of each Dockerfile stage
+        self.dfp = df_parser(self.workflow.builder.df_path)
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_hide_files.py]: I may be wrong, but it seems that `self.dfp` is initialized out of `__init__`.  Is actually dfp needed as object attribute, could be just variable?

File: atomic_reactor/plugins/pre_hide_files.py [hunk 2]
```diff
@@ -43,8 +43,28 @@ def run(self):
         self._populate_start_file_lines()
         self._populate_end_file_lines()
 
-        for parent_image_id in self.workflow.builder.parent_images.values():
-            self._update_dockerfile(parent_image_id)
+        # Find limits of each Dockerfile stage
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_hide_files.py]: Could be this whole block extracted into method like `_update_build_stages` to have `run` written as steps on the same level of abstraction

File: atomic_reactor/plugins/pre_hide_files.py (hunk 1 above)
Comment (MartinBasti) [reply atomic_reactor/plugins/pre_hide_files.py]: oh, found it. But still it should be defined in `__init__` as None at least

File: atomic_reactor/plugins/pre_hide_files.py (hunk 2 above)
Comment (twaugh) [reply atomic_reactor/plugins/pre_hide_files.py]: Good idea. Pushed a change, see what you think.

File: atomic_reactor/plugins/pre_hide_files.py (hunk 1 above)
Comment (MartinBasti) [reply atomic_reactor/plugins/pre_hide_files.py]: (lazy evaluated property would be the best)

Comment (twaugh) [reply atomic_reactor/plugins/pre_hide_files.py]: Thanks, updated.
It has to be None in `__init__` because an earlier plugin modifies the file.

Comment (twaugh) [reply atomic_reactor/plugins/pre_hide_files.py]: I think we want to explicitly parse during `run()` to make sure we don't parse too early.

Comment (MartinBasti) [reply atomic_reactor/plugins/pre_hide_files.py]: lazy property should be parsed at first usage, that should happen somewhere inside `run()`, but I'm fine with `None`

**Comment by twaugh** [comment]
No release notes update needed.

**Pull request merged**